aptos-move/framework/aptos-framework/sources/configs/gas_schedule.move

```diff
@@ -13,6 +14,13 @@ module aptos_framework::gas_schedule {
 
     /// The provided gas schedule bytes are empty or invalid
     const EINVALID_GAS_SCHEDULE: u64 = 1;
+    const EINVALID_GAS_FEATURE_VERSION: u64 = 2;
+
+    const GENESIS_GAS_FEATURE_VERSION: u64 = 0;
+
+    struct GasFeatureVersion has key, copy, drop {
```

Comment:
This is very concerning being stored here as a single resource, given the known costs of small resources. @msmouse won't like this. Can't this be piggy-packed with the existing gas schedule resource?

I know that in this case you need to introduce a `GasScheduleV2` and implement the upgrade logic, but this maybe worth the cost, considering that this resource will be needed about _every_ time you calculate gas.

Reply:
> This is very concerning being stored here as a single resource, given the known costs of small resources.

This is a fair point. Although I feel like it may not matter that much for this particular case since this is just a system resource that's also a singleton.

>  introduce a GasScheduleV2

Nonetheless I actually like this idea. Let me see how I can implement the upgrade logic.